docs/docs/guides/03-typescript.md

```diff
@@ -2,4 +2,266 @@
 title: Typescript
 ---
 
-🚧 coming soon 🚧
+# Development Guidelines
+
+This document provides guidelines for developing on the Strapi codebase using TypeScript. It covers various topics, including types location, generics, breaking changes, factories and more.
+
+## 1. Type Location
+
+When working with types on the Strapi codebase, it is essential to consider the location of types. These are the three common approaches and situations we document:
+
+### a. Colocation
+
+In the colocation approach, type definitions are colocated with the corresponding code. This means that types are defined in the same file or module where they are used.
+
+**This approach is suitable when the types are specific to a particular module**.
+
+_Note: In some situations, colocated types can still be exported if needed by other modules (e.g. React Component Props)_
+
+Example:
+
+```typescript
+// config.ts
+
+interface Config {
+  prop1: string;
+  prop2: number;
+}
+
+function update(config: Config): Config {
+  // ...
+}
+```
+
+```typescript
+// Box.tsx
+
+import * as React from 'react';
+
+export interface TaskProps {
+  name: string;
+  priority: 'high' | 'medium' | 'low';
+  due: Date;
+}
+
+export const TaskProps = ({ name, priority, due }: TaskProps) => {
+  // ...
+};
+```
+
+### b. Definitions Files
+
+**When types are shared across multiple files or modules, or when dealing with types that does not belong to a specific module**, it is recommended to use definition files (`.d.ts`) to separate type declarations from their usage.
```

Comment:
🤔  do you have resource explaining the pros&cons of using definition files instead of let's say a `types.ts` file ? for shared types in a folder ?

Reply:
Well yup +1